### ISSUE TITLE
If a relocating venv doesn't have "user packages", `pip install` without packages names throws an error

### ISSUE THREAD

**Issue by carlosperate** [mu-editor/mu#1651]
Noticed this in the my logs, it didn't seem to break the process as it continue successfully, but we should look into this:
```
2021-06-14 13:01:35,434 - mu.virtual_environment:548(recreate) DEBUG: About to reinstall user packages: []
2021-06-14 13:01:35,434 - mu.virtual_environment:892(install_user_packages) INFO: Installing user packages: 
2021-06-14 13:01:35,435 - mu.virtual_environment:105(run_blocking) INFO: About to run blocking /Users/microbit-carlos/Library/Application Support/mu/mu_venv-38-20210614-130036/bin/pip with args ['install', '--disable-pip-version-check', '--upgrade'] and envvars {}
2021-06-14 13:01:36,222 - mu.virtual_environment:147(wait) DEBUG: Finished: True; exitStatus 0; exitCode 1
2021-06-14 13:01:36,222 - mu.virtual_environment:168(wait) ERROR: ERROR: You must give at least one requirement to install (see "pip help install")
2021-06-14 13:01:36,224 - mu.virtual_environment:619(ensure_and_create) ERROR: Process finished but with error code 1:
ERROR: You must give at least one requirement to install (see "pip help install")
2021-06-14 13:01:36,225 - mu.virtual_environment:517(quarantine_venv) INFO: Quarantined /Users/microbit-carlos/Library/Application Support/mu/mu_venv-38-20210614-130036 as /Users/microbit-carlos/Library/Application Support/mu/mu_venv-38-20210614-130036.FAILED
2021-06-14 13:01:36,225 - mu.virtual_environment:443(relocate) DEBUG: Relocating to /Users/microbit-carlos/Library/Application Support/mu/mu_venv-38-20210614-130136
2021-06-14 13:01:36,226 - mu.virtual_environment:459(relocate) DEBUG: Virtual environment set up mu_venv-38-20210614-130136 at /Users/microbit-carlos/Library/Application Support/mu/mu_venv-38-20210614-130136
```

So we detect the user packages at the top of `VirtualEnvironment.recreate()` and reinstalled right at the end:
https://github.com/mu-editor/mu/blob/6d3f3b9e19ab3c739f2610b8956d6fe7262ca9fa/mu/virtual_environment.py#L527-L549

Detecting the user packages is done by iterating over all the packages detected and excluding those that are not part of the "baseline" packages, so we can get an empty list:
https://github.com/mu-editor/mu/blob/6d3f3b9e19ab3c739f2610b8956d6fe7262ca9fa/mu/virtual_environment.py#L913-L920

We can check if the list is empty at `VirtualEnvironment.install_user_packages()`, it's a very small method:

https://github.com/mu-editor/mu/blob/6d3f3b9e19ab3c739f2610b8956d6fe7262ca9fa/mu/virtual_environment.py#L888-L894

**Comment by carlosperate** [comment]
Should be resolved by https://github.com/mu-editor/mu/pull/1650

**Comment by carlosperate** [comment]
This should be resolved by #1650.
